packages/framework/src/Support/Filesystem/MediaFile.php

```diff
@@ -27,6 +29,17 @@ class MediaFile extends ProjectFile
     /** @var array<string> The default extensions for media types */
     final public const EXTENSIONS = ['png', 'svg', 'jpg', 'jpeg', 'gif', 'ico', 'css', 'js'];
 
+    public function __construct(string $path)
+    {
+        $path = trim_slashes(Str::after(Hyde::pathToRelative($path), Hyde::getMediaDirectory()));
+
+        if (Filesystem::missing($path)) {
+            throw new FileNotFoundException($path);
+        }
```

Comment:
This should not be possible for this to happen with normal usage, since we generate these based on discovered files. If a developer does something custom, we can't assume they want files backed by disk. We'll see how we handle construct time data generation though...

It also makes unit testing harder (we could add an internal static property to disable validation, or mock filesystem)

Reply:
We could add validation to the static constructor. But that discrepancy adds more complexity than static property. Filesystem mocks add test complexity